### PR TITLE
Cuda configuration option

### PR DESCRIPTION
**Pull request by DanilBaibak**
Add a way to disble cuda configurations when using generate-binary-build-matrix

### CODE REVIEW

**Comment by weiwangmeta** [comment]
what would fail or what is the inconvenience of not having this PR?

**Review by atalman — COMMENTED** [review]
Why we want to do this can you please explain ? Maybe easier way to get full matrix and filter pick cpu only configs ?

**Comment by osalpekar** [comment]
@weiwangmeta @atalman Context is here: https://github.com/pytorch/test-infra/issues/744. In summary, pytorch/text does not need CUDA builds for Linux and Windows, so we should account for that when we generate the build matrix.

However, this seems to have disabled CUDA for all Linux builds (for example, no Linux CUDA builds for audio and vision have been triggered in the checks, which we still need). Would changing the order of things work? As in we start generating the matrix in the build workflows (as the first step) rather than in the test workflow, and the test workflow sets the `with-cuda` flag for each repo (just like we set pre/post script). Then the build workflow gets the `with-cuda` arg as input, generates the appropriate matrix, and kicks off the builds. @seemethere IIRC we were planning to move matrix generation to the build workflows eventually anyways?

The other option could be filter out the non-cpu builds per-repo as @atalman mentioned.

File: .github/workflows/test_build_conda_macos.yml [hunk 1]
```diff
@@ -22,8 +22,8 @@ jobs:
       matrix:
         include:
           - repository: pytorch/audio
-            pre-script: packaging/pre_build_script_wheel.sh
-            post-script: packaging/post_build_script_wheel.sh
+            pre-script: packaging/pre_build_script.sh
```
Comment (osalpekar) [inline .github/workflows/test_build_conda_macos.yml:25]: You can rebase now that https://github.com/pytorch/test-infra/pull/750 is merged, and these script-name changes shouldn't be needed anymore!

**Comment by DanilBaibak** [comment]
@atalman looks a bit clearer to me when you can run generate-matrix with specific arguments and get the required matrix.

**Comment by DanilBaibak** [comment]
@osalpekar good point - we still need cuda for vision and audio for linux build! I created two separate test builds for linux with and without cuda.

**Review by atalman — APPROVED** [review]
lgtm

**Pull request merged**